File: 2260-divide-a-string-into-groups-of-size-k/divide-a-string-into-groups-of-size-k.py
class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
        result = []
        n = len(s)
        i = 0
        while i < n:
            char = s[i:i+k]
            if len(char) < k:
                num = k - len(char)
                char += fill * num
            result.append(char)
            i += k
        return result

File: 2260-divide-a-string-into-groups-of-size-k/divide-a-string-into-groups-of-size-k.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        vector<string> result;
        int n = s.length();
        int i = 0;
        while (i < n) {
            string group = s.substr(i, k);
            if (group.length() < k) {
                group += string(k - group.length(), fill);
            }
            result.push_back(group);
            i += k;
        }
        return result;
    }
};